features/step_definitions/developer_portal/base.rb

```diff
@@ -0,0 +1,9 @@
+When "they visit the developer portal" do
+  set_current_domain(@provider.domain)
+  visit '/'
+end
```

Comment:
This one exists already: https://github.com/3scale/porta/blob/f0cca55c5e9ade1f5219eadea7d7d36df0201b1b/features/step_definitions/url_steps.rb#L7-L10

Can also reuse:
```cucumber
When the current domain is foo.3scale.localhost
And they visit the homepage
```

From https://github.com/3scale/porta/blob/f0cca55c5e9ade1f5219eadea7d7d36df0201b1b/features/support/paths.rb#L23-L24

Perhaps add a new key:
```ruby
when /the home\s?page/, 'the developer portal'
      '/'
```

```cucumber
When the current domain is foo.3scale.localhost
And they visit the developer portal
```

If you still prefer to use this custom step, add it to `custom_web_steps` to keep better track of the duplication: https://github.com/3scale/porta/blob/f0cca55c5e9ade1f5219eadea7d7d36df0201b1b/features/step_definitions/custom_web_steps.rb#L175-L177